blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -60,11 +60,13 @@ object BlazeClient {
             .handleError(e => logger.error(e)("Error invalidating connection"))
 
         def borrow =
-          Resource.makeCase(manager.borrow(key)) {
-            case (_, ExitCase.Completed) =>
-              F.unit
-            case (next, ExitCase.Error(_) | ExitCase.Canceled) =>
+        // TODO: The `attempt` here is a workaround for https://github.com/typelevel/cats-effect/issues/487 .
+        //  It can be removed once the issue is resolved.
+          Resource.makeCase(manager.borrow(key).attempt) {
+            case (Right(next), ExitCase.Error(_) | ExitCase.Canceled) =>
               invalidate(next.connection)
+            case _ =>
```

Comment:
Could you please explain, @RafalSumislawski, why it's not necessary to explicitly handle the `case (Left(error), _)) => ...` case?

In other words, what's the import of returning `IO.unit` if, as I understand, `manager.borrow(key).attempt` returns an `F(Left(RuntimeException(...)))`?

Reply:
The lines 65-70 are the `release` function. If acquiring the connection failed, there's nothing to release. There's also no need to propagate the error here, as it will already be propagated by `case Left(t) => F.raiseError(t)` on line 88. So we do nothing aka return `IO.unit`.